Update discoverers table to account for elements that were missing discoverers

heyitskei opened the pull request.

heyitskei committed:
Update discoverers table to account for elements that were missing discoverers

BrayonCrayon committed:
Ensure no empty names for discoverers are entered into the db

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

heyitskei committed:
Update created_at and updated_at in the discoverers table

heyitskei committed:
Update created_at and updated_at in the types table, filter out invalid names

heyitskei committed:
WIP - determined that a type can be nullable; need to adjust db structure

heyitskei committed:
Update Types in the CSV to ensure no element is missing a type

Co-authored-by: Brady Deroy <b.deroy@vehikl.com>

heyitskei committed:
WIP - Update Discoverers in the CSV to standardize how multiple people are stored as discoverers on elements

Co-authored-by: Brady Deroy <b.deroy@vehikl.com>

BrayonCrayon committed:
Updated element discoveries with more discoverers that were abbreviated with "et al" or missed

BrayonCrayon committed:
Update test and job implementation to account for multiple discoverers separated by commas

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Kim Gallego <kimberly.gallego@vehikl.com>

BrayonCrayon committed:
start reworking the update discoverers table migration to account for existing discoverers before linking them with the elements
-WIP

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

BrayonCrayon committed:
Finish migration

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>

BrayonCrayon committed:
Clean up

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>

BrayonCrayon commented on app/Jobs/ImportElementDataJob.php:
```diff
@@ -27,18 +28,37 @@ public function handle(): void
+                'created_at' => Carbon::now(),
+                'updated_at' => Carbon::now(),
+            ])
+            ->filter(function ($discoverer) {
+                return $discoverer['name'] !== '';
+            })
```
refactor to use reject

BrayonCrayon replied:
We changed it to be a reject instead

BrayonCrayon commented on app/Jobs/ImportElementDataJob.php:
```diff
@@ -70,6 +90,9 @@ public function handle(): void
 
     public function rowToElementInsert($elementStates, $elementTypes, $row): array //
     {
+        if ($elementTypes->first(fn ($item) => $item->name === $row['Type'])?->id === null) {
+            dump($elementStates->toArray(), $elementTypes->toArray(), $row);
+        }
```
this has gotta go

BrayonCrayon replied:
We have removed it

BrayonCrayon commented on app/Models/ElementState.php:
```diff
@@ -10,6 +10,7 @@ class ElementState extends Model
     protected $fillable = [
-      'name'
+      'name',
+        'created_at','updated_at'
```
fix this formatting

BrayonCrayon replied:
Fixed the formatting

BrayonCrayon committed:
PR Feedback
- Stay consistent with using reject on values that do not exist in job
- Remove dump
- Update formatting in model

Co-authored-by: Ekaterina Stroganova <e.stroganova@vehikl.com>
Co-authored-by: Brandon Cormier <b.cormier@vehikl.com>
